pkg/file/seekjoiner/internal/job.go

```diff
@@ -21,18 +21,20 @@ type SimpleJoinerJob struct {
 	spanLength int64
 	off        int64
 	levels     int
+	refLength  int
 
 	ctx    context.Context
 	getter storage.Getter
 }
 
 // NewSimpleJoinerJob creates a new simpleJoinerJob.
-func NewSimpleJoinerJob(ctx context.Context, getter storage.Getter, rootChunk swarm.Chunk) *SimpleJoinerJob {
+func NewSimpleJoinerJob(ctx context.Context, getter storage.Getter, refLength int, rootChunk swarm.Chunk) *SimpleJoinerJob {
```

Comment:
refLength can be measured from rootChunk here. No need to pass both rootChunk and refLength.

Reply:
well not entirely, because the root chunk we have here is the root chunk that came from a retrieve request. when we retrieve the root chunk of encrypted content we retrieve the content addressed chunk, we don't concatenate the decryption key to it. i'd leave it as is for now

Reply:
You are saying that the refLength there does not represent the length of the rootChunk.Address()? I see that this function is used in https://github.com/ethersphere/bee/pull/582/files#diff-f069298b920c3d61a6ec16bace277a31R63 and it would be very strange that address and rootChunk.Address() are different in simpleJoiner.Join function where getter.Get is used.

I think that it need to be clarified what is refLength is expected to be if it is different then rootChunk.Address().

Reply:
so... the problem is the following:
you get a request to join the whole thing, which is address+key, but you need the root chunk, so you ask it from the retrieval protocol, which brings it back, but it does not bring it back with the key, since it is already decrypted. so you get back from the retrieve protocol the chunk with the content addressed hash of the encrypted content (no key) and the content. i am a bit afraid to augment the key into the chunk decryption key along with the decrypted content since it kind of takes away the purity of the chunk type and adds more information to it, since in this case Address is not just Address but Address+Key, which is kind of not so great. I can intercept the chunk before it is sent to the joiner and create a new chunk, and augment the decryption key there, but i find this a bit ugly